Content.Server/GameTicking/Rules/RevolutionaryRuleSystem.cs

```diff
@@ -0,0 +1,463 @@
+    {
+        if (HasComp<HeadRevolutionaryComponent>(mind.OwnedEntity))
+            return;
+
+        var query = AllEntityQuery<RevolutionaryRuleComponent, GameRuleComponent>();
+        while (query.MoveNext(out var uid, out var comp, out var gameRule))
+        {
+            _mindSystem.AddRole(mind, new RevolutionaryRole(mind, _prototypeManager.Index<AntagPrototype>(comp.HeadRevPrototypeId)));
+            var inCharacterName = string.Empty;
+            if (mind.OwnedEntity != null)
+            {
+                if (HasComp<CommandStaffComponent>(mind.OwnedEntity.Value))
+                {
+                    RemComp<CommandStaffComponent>(mind.OwnedEntity.Value);
+                }
+                EnsureComp<HeadRevolutionaryComponent>(mind.OwnedEntity.Value);
+                EnsureComp<RevolutionaryComponent>(mind.OwnedEntity.Value);
+                if (_inventory.TryGetSlotContainer(mind.OwnedEntity.Value, "back", out var containerSlot, out var slotDefinition))
+                {
+                    var bag = containerSlot.ContainedEntity;
+                    var flash = Spawn("Flash", new EntityCoordinates(mind.OwnedEntity.Value, Vector2.Zero));
+                    var glasses = Spawn("ClothingEyesGlassesSunglasses", new EntityCoordinates(mind.OwnedEntity.Value, Vector2.Zero));
+                    if (bag != null && _storageSystem.CanInsert((EntityUid) bag, flash, out var reason))
+                    {
+                        _storageSystem.Insert((EntityUid) bag, flash);
+                        _storageSystem.Insert((EntityUid) bag, glasses);
+                    }
+                    else
+                    {
+                        //Honestly if they don't have a bag they are just going to take this L.
+                        _entityManager.DeleteEntity(flash);
+                        _entityManager.DeleteEntity(glasses);
+                        _stationSpawningSystem.EquipStartingGear(mind.OwnedEntity.Value, _prototypeManager.Index<StartingGearPrototype>(comp.HeadRevGearPrototypeId), null);
+                    }
+                }
+                _npcFaction.RemoveFaction(mind.OwnedEntity.Value, "NanoTrasen", false);
+                _npcFaction.AddFaction(mind.OwnedEntity.Value, "Revolutionary");
+                inCharacterName = MetaData(mind.OwnedEntity.Value).EntityName;
+                comp.HeadRevs.Add(inCharacterName, headRev.Name);
+                _audioSystem.PlayGlobal(comp.HeadRevStartSound, mind.OwnedEntity.Value);
+            }
+            if (mind.Session != null)
+            {
+                var message = Loc.GetString("head-rev-role-greeting");
+                var wrappedMessage = Loc.GetString("chat-manager-server-wrap-message", ("message", message));
+                _chatManager.ChatMessageToOne(Shared.Chat.ChatChannel.Server, message, wrappedMessage, default, false, mind.Session.ConnectedClient, Color.FromHex("#5e9cff"));
+            }
+        }
```

Comment:
Don't hardcode in a bunch of things here, make it a re-useable method. Most of this is just ignoring code we already have (also removing NanoTrasen means people will be able to meta this so please remove).

You shouldn't be hardcoding in flash and glass spawns, especially considering these can be metagamed. Put this into a general role-assign method and make nukies use it as this will quickly become unmaintainble.